src/components/TrailForm.js

```diff
@@ -0,0 +1,169 @@
+import React from 'react';
+import { withStyles } from 'material-ui/styles';
+import Card, { CardActions, CardContent } from 'material-ui/Card';
+import Button from 'material-ui/Button';
+import Typography from 'material-ui/Typography';
+import TextField from 'material-ui/TextField';
+import Delete from '@material-ui/icons/Delete';
+import Select from 'material-ui/Select';
+import { MenuItem } from 'material-ui/Menu';
+import { FormControl, FormHelperText } from 'material-ui/Form';
+import _ from 'lodash';
+import Input, { InputLabel } from 'material-ui/Input';
+import Collapse from 'material-ui/transitions/Collapse';
+import List, { ListItem, ListItemIcon, ListItemText } from 'material-ui/List';
+import ExpandLess from '@material-ui/icons/ExpandLess';
+import ExpandMore from '@material-ui/icons/ExpandMore';
+import { SketchPicker } from 'react-color';
+import { getMapStyle } from './../utils/mapUtils';
+import HydrantList from './HydrantList';
+import ColorPicker from './ColorPicker';
+import Close from '@material-ui/icons/Close';
+
+const styles = theme => ({
+  root: {
+    width: '100%',
+  },
+  input: {
+    fontSize: '1.5rem',
+  },
+  textField: {
+    marginLeft: '1em',
+    marginRight: '1em',
+    width: 200,
+    display: 'block',
+  },
+  inset: {
+    paddingLeft: 0
+  },
+  pos: {
+    marginBottom: 12,
+  },
+  nested: {
+    paddingLeft: theme.spacing.unit * 4,
+  },
+  mini: {
+    boxShadow: '0px -1px 6px 6px rgba(0, 0, 0, 0.29), 0px 6px 10px 0px rgba(0, 0, 0, 0.14), 0px 1px 18px 0px rgba(0, 0, 0, 0.12)'
+  }
+});
+
+
+class TrailForm extends React.Component {
+
+  state = {
+    trailSectionsOpen: true,
+    pickerOpen: false,
+    trailFill: 'white',
+    }
+
+  render() {
+    const {
+      classes,
+      modifyTrail,
+      editableTrail,
+      trailEditable,
+      interaction,
+      interactionChanged,
+      hydrants,
+      hydrantDeleted,
+      modifyHydrant
+    } = this.props;
+
+    const isTrailMode = interaction === 'DRAW_MODIFY_TRAIL'
+
+    const highlightPoly = (feature) => {
```

Comment:
This is a nitpick, but can we rename these to `(un)highlightFeature`? Polygon's are specific geometries inside of features (which can potentially contain many geometries/polygons), better to avoid any future confusion looking back at this later if we have some features with multiple shapes inside them.

Reply:
Or can just switch to `toggleHighlight` style like the other component uses.

Reply:
Done. Check.